crates/swc_ecma_parser/src/parser/input.rs

```diff
@@ -143,15 +141,14 @@ impl Tokens for TokensInput {
     fn take_errors(&mut self) -> Vec<Error> {
         take(&mut self.errors.borrow_mut())
     }
-
-    fn reset_to(&mut self, _: BytePos) {}
 }
 
 /// Note: Lexer need access to parser's context to lex correctly.
 #[derive(Debug)]
 pub struct Capturing<I: Tokens> {
     inner: I,
-    captured: Rc<RefCell<Vec<TokenAndSpan>>>,
+    // use a buffered singly linked list because clones occur infrequently
+    captured: BufferedSinglyLinkedList<TokenAndSpan>,
```

Comment:
I have yet to profile whether `BufferedSinglyLinkedList`, `SinglyLinkedList`, or something else would be more appropriate.